### ISSUE TITLE
Empty Security service injected to MediaDataProvider

### ISSUE THREAD

**Issue by StephenBeirlaen** [sulu/sulu#5587]
| Q | A
| --- | ---
| Sulu 2.2.0 | 2dac9dd6930146186301ff9f6b5a7d84921a8dc2
| Chrome | 85.0.4183.121 (Official Build) (64-bit)

# Actual Behavior

I just updated to Sulu 2.2.0 from 2.1 (93413d73815412d3adc9658ff31e450c5de028c1). All pages of the website (not admin) now fail with the following exception:

`Argument 6 passed to Sulu\Component\Media\SmartContent\MediaDataProvider::__construct() must be an instance of Symfony\Component\Security\Core\Security, null given, called in /.../var/cache/website/dev/ContainerRwFhJjS/getSuluPage_SmartContent_DataProviderPoolService.php on line 45`

The Security argument has been added a few days ago in https://github.com/sulu/sulu/pull/5560, so I thought this may have been an oversight. Shouldn't this argument be nullable? Or is there an upgrade guide that I am missing?

I can fix this error by uncommenting the default security configuration from the skeleton: https://github.com/sulu/skeleton/blob/master/config/packages/security_website.yaml (this file was empty in the past, that's why I had no configuration here).

```
# To enable security bundle for website you need to change `config/bundles.php` file

security:
    encoders:
        Sulu\Bundle\SecurityBundle\Entity\User: bcrypt

    providers:
        sulu:
            id: sulu_security.user_provider

    firewalls:
        website:
            pattern: ^/
            anonymous: lazy
            # The login and logout routes need to be created.
            # For an advanced user management with registration and opt-in emails have a look at the:
            # https://github.com/sulu/SuluCommunityBundle
            form_login:
                login_path: login
                check_path: login
            logout:
                path: logout
                target: /
            remember_me:
                secret:   "%kernel.secret%"
                lifetime: 604800 # 1 week in seconds
                path:     /

sulu_security:
    checker:
        enabled: true
```

To make this configuration work I also had to change my bundles.php (as stated in the comment above):
from
`Symfony\Bundle\SecurityBundle\SecurityBundle::class => ['all' => true, 'admin' => true],`
to
`Symfony\Bundle\SecurityBundle\SecurityBundle::class => ['all' => true],`

# Expected Behavior

Sulu should work even if security is not configured for the website

# Steps to Reproduce

- Install a Sulu skeleton while NOT uncommenting website security: https://github.com/sulu/skeleton/blob/master/config/packages/security_website.yaml
- Add a page, the exception occurs in the admin preview and in the website

**Comment by nnatter** [comment]
Hey, thanks for reporting! This was already fixed by #5580 and will hopefully be released soon 🙂

**Comment by StephenBeirlaen** [comment]
Great! Did not see that one. This can be closed then.
